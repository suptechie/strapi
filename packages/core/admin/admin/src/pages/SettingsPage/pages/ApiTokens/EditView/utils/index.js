import getActionsState from './getActionsState';
import getDateOfExpiration from './getDateOfExpiration';
import schema from './schema';
import togglePermissions from './togglePermissions';

export { getActionsState, getDateOfExpiration, schema, togglePermissions };
