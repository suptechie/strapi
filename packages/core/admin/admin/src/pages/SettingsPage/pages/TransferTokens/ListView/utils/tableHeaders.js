const tableHeaders = [
  {
    name: 'name',
    key: 'name',
    metadatas: {
      label: {
        id: 'Settings.transferTokens.ListView.headers.name',
        defaultMessage: 'Name',
      },
      sortable: true,
    },
  },
  {
    name: 'description',
    key: 'description',
    metadatas: {
      label: {
        id: 'Settings.transferTokens.ListView.headers.description',
        defaultMessage: 'Description',
      },
      sortable: false,
    },
  },
  {
    name: 'createdAt',
    key: 'createdAt',
    metadatas: {
      label: {
        id: 'Settings.transferTokens.ListView.headers.createdAt',
        defaultMessage: 'Created at',
      },
      sortable: false,
    },
  },
  {
    name: 'lastUsedAt',
    key: 'lastUsedAt',
    metadatas: {
      label: {
        id: 'Settings.transferTokens.ListView.headers.lastUsedAt',
        defaultMessage: 'Last used',
      },
      sortable: false,
    },
  },
];

export default tableHeaders;
