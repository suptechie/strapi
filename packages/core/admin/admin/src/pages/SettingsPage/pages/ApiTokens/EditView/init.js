import { transformPermissionsData } from './utils';

const init = (state, permissions = []) => {
  return {
    ...state,
    selectedActions: [],
    data: transformPermissionsData(permissions),
  };
};

export default init;
