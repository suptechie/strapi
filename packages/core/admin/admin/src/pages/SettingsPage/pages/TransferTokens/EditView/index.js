import React, { useState } from 'react';
import { useIntl } from 'react-intl';
import { Formik } from 'formik';
import { useRouteMatch, useHistory } from 'react-router-dom';
import { useQuery } from 'react-query';
import {
  SettingsPageTitle,
  useFocusWhenNavigate,
  Form,
  useOverlayBlocker,
  useNotification,
  useGuidedTour,
  useRBAC,
  useFetchClient,
} from '@strapi/helper-plugin';
import { Main } from '@strapi/design-system/Main';
import { Stack } from '@strapi/design-system/Stack';
import { ContentLayout } from '@strapi/design-system/Layout';
import { formatAPIErrors } from '../../../../../utils';
import { schema } from './utils';
import LoadingView from './components/LoadingView';
import adminPermissions from '../../../../../permissions';
import FormTransferTokenContainer from './components/FormTransferTokenContainer';
import TokenBox from '../../../components/Tokens/TokenBox';
import FormHead from '../../../components/Tokens/FormHead';

const MSG_ERROR_NAME_TAKEN = 'Name already taken';

const TransferTokenCreateView = () => {
  useFocusWhenNavigate();
  const { formatMessage } = useIntl();
  const { lockApp, unlockApp } = useOverlayBlocker();
  const toggleNotification = useNotification();
  const history = useHistory();
  const [transferToken, setTransferToken] = useState(
    history.location.state?.transferToken.accessKey
      ? {
          ...history.location.state.transferToken,
        }
      : null
  );
  const { setCurrentStep } = useGuidedTour();
  const {
    allowedActions: { canCreate, canUpdate, canRegenerate },
  } = useRBAC(adminPermissions.settings['transfer-tokens']);
  const {
    params: { id },
  } = useRouteMatch('/settings/transfer-tokens/:id');
  const { get, post, put } = useFetchClient();

  const isCreating = id === 'create';

  const { status } = useQuery(
    ['transfer-token', id],
    async () => {
      const {
        data: { data },
      } = await get(`/admin/transfer/tokens/${id}`);

      setTransferToken({
        ...data,
      });

      return data;
    },
    {
      enabled: !isCreating && !transferToken,
      onError() {
        toggleNotification({
          type: 'warning',
          message: { id: 'notification.error', defaultMessage: 'An error occured' },
        });
      },
    }
  );

  const handleSubmit = async (body, actions) => {
    lockApp();
    const lifespanVal =
      body.lifespan && parseInt(body.lifespan, 10) && body.lifespan !== '0'
        ? parseInt(body.lifespan, 10)
        : null;

    try {
      const {
        data: { data: response },
      } = isCreating
        ? await post(`/admin/transfer/tokens`, {
            ...body,
            lifespan: lifespanVal,
            permissions: ['push'],
          })
        : await put(`/admin/transfer/tokens/${id}`, {
            name: body.name,
            description: body.description,
            type: body.type,
            permissions: ['push'],
          });

      unlockApp();

      if (isCreating) {
        history.replace(`/settings/transfer-tokens/${response.id}`, { transferToken: response });
        setCurrentStep('transferTokens.success');
      }
      setTransferToken({
        ...response,
      });

      toggleNotification({
        type: 'success',
        message: isCreating
          ? formatMessage({
              id: 'notification.success.transfertokencreated',
              defaultMessage: 'Transfer Token successfully created',
            })
          : formatMessage({
              id: 'notification.success.transfertokenedited',
              defaultMessage: 'Transfer Token successfully edited',
            }),
      });
    } catch (err) {
      const errors = formatAPIErrors(err.response.data);
      actions.setErrors(errors);

      if (err?.response?.data?.error?.message === MSG_ERROR_NAME_TAKEN) {
        toggleNotification({
          type: 'warning',
          message: err.response.data.message || 'notification.error.tokennamenotunique',
        });
      } else {
        toggleNotification({
          type: 'warning',
          message: err?.response?.data?.message || 'notification.error',
        });
      }
      unlockApp();
    }
  };

  const canEditInputs = (canUpdate && !isCreating) || (canCreate && isCreating);
  const isLoading = !isCreating && !transferToken && status !== 'success';

  if (isLoading) {
    return <LoadingView transferTokenName={transferToken?.name} />;
  }

  return (
    <Main>
      <SettingsPageTitle name="Transfer Tokens" />
      <Formik
        validationSchema={schema}
        validateOnChange={false}
        initialValues={{
          name: transferToken?.name || '',
          description: transferToken?.description || '',
          lifespan: transferToken?.lifespan
            ? transferToken.lifespan.toString()
            : transferToken?.lifespan,
        }}
        enableReinitialize
        onSubmit={(body, actions) => handleSubmit(body, actions)}
      >
        {({ errors, handleChange, isSubmitting, values }) => {
          return (
            <Form>
              <FormHead
                backUrl="/settings/transfer-tokens"
                title={{
                  id: 'Settings.transferTokens.createPage.title',
                  defaultMessage: 'Create Transfer Token',
                }}
                token={transferToken}
                setToken={setTransferToken}
                canEditInputs={canEditInputs}
                canRegenerate={canRegenerate}
                isSubmitting={isSubmitting}
                regenerateUrl="/admin/transfer/tokens/"
              />
              <ContentLayout>
                <Stack spacing={6}>
                  {Boolean(transferToken?.name) && <TokenBox token={transferToken?.accessKey} />}
                  <FormTransferTokenContainer
                    errors={errors}
                    onChange={handleChange}
                    canEditInputs={canEditInputs}
                    isCreating={isCreating}
                    values={values}
                    transferToken={transferToken}
                  />
                </Stack>
              </ContentLayout>
            </Form>
          );
        }}
      </Formik>
    </Main>
  );
};

export default TransferTokenCreateView;
