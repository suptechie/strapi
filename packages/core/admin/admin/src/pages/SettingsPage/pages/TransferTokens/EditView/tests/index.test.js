import React from 'react';

import { render, waitFor } from '@tests/utils';

import EditView from '../index';

jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  useRBAC: jest.fn(() => ({
    allowedActions: {
      canCreate: true,
      canDelete: true,
      canRead: true,
      canUpdate: true,
      canRegenerate: true,
    },
  })),
}));

describe('ADMIN | Pages | TRANSFER TOKENS | EditView', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders and matches the snapshot when editing existing token', async () => {
    const { getByText } = render(<EditView />, {
      initialEntries: ['/settings/transfer-tokens/1'],
    });

    await waitFor(() => expect(getByText('My super token')).toBeInTheDocument());
    expect(getByText('This describe my super token')).toBeInTheDocument();
    expect(getByText('Regenerate')).toBeInTheDocument();
  });
});
