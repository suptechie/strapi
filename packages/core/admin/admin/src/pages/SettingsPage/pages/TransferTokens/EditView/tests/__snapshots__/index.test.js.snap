// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADMIN | Pages | TRANSFER TOKENS | EditView renders and matches the snapshot when creating token 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <div
    class="c0"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`ADMIN | Pages | TRANSFER TOKENS | EditView renders and matches the snapshot when editing existing token 1`] = `
.c6 {
  min-width: 0;
}

.c11 {
  background: #4945ff;
  padding: 8px;
  padding-right: 16px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c16 {
  padding-right: 56px;
  padding-left: 56px;
}

.c1 {
  background: #f6f6f9;
  padding-top: 24px;
  padding-right: 56px;
  padding-bottom: 40px;
  padding-left: 56px;
}

.c2 {
  padding-bottom: 8px;
}

.c18 {
  background: #ffffff;
  padding: 24px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c19 {
  background: #f6f6f9;
  padding: 12px;
  border-radius: 4px;
}

.c25 {
  background: #ffffff;
  padding-top: 24px;
  padding-right: 32px;
  padding-bottom: 24px;
  padding-left: 32px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c36 {
  border-radius: 4px;
  border-color: #dcdce4;
  border: 1px solid #dcdce4;
}

.c38 {
  font-size: 0.875rem;
  background: #ffffff;
  color: #32324d;
  padding: 16px;
  border-radius: 4px;
  width: 100%;
  height: 6.5625rem;
  line-height: 1.43;
}

.c40 {
  background: #eaeaef;
  padding-right: 12px;
  padding-left: 12px;
  border-radius: 4px;
  position: relative;
  overflow: hidden;
  width: 100%;
  cursor: default;
}

.c43 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c50 {
  background: #ffffff;
  padding-right: 12px;
  padding-left: 12px;
  border-radius: 4px;
  position: relative;
  overflow: hidden;
  width: 100%;
  cursor: default;
}

.c9 {
  font-weight: 600;
  font-size: 2rem;
  line-height: 1.25;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #32324d;
}

.c14 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  line-height: 0;
  color: #ffffff;
}

.c22 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 500;
  color: #32324d;
}

.c24 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #666687;
}

.c27 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.25;
  color: #32324d;
}

.c30 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c32 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #d02b20;
}

.c45 {
  font-size: 0.875rem;
  line-height: 1.43;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #666687;
}

.c49 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #666687;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c17 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 24px;
}

.c21 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

.c26 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
}

.c41 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 16px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c44 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 12px;
}

.c12 {
  position: relative;
  outline: none;
}

.c12 > svg {
  height: 12px;
  width: 12px;
}

.c12 > svg > g,
.c12 > svg path {
  fill: #ffffff;
}

.c12[aria-disabled='true'] {
  pointer-events: none;
}

.c12:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c12:focus-visible {
  outline: none;
}

.c12:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c51 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c52[aria-disabled='true'] .c8 {
  color: #666687;
}

.c52[aria-disabled='true']:active .c8 {
  color: #666687;
}

.c13 {
  height: 2rem;
  border: 1px solid #dcdce4;
  background: #ffffff;
}

.c13[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c13[aria-disabled='true'] .c8 {
  color: #666687;
}

.c13[aria-disabled='true'] svg > g,.c13[aria-disabled='true'] svg path {
  fill: #666687;
}

.c13[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c13[aria-disabled='true']:active .c8 {
  color: #666687;
}

.c13[aria-disabled='true']:active svg > g,.c13[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c13:hover {
  background-color: #f6f6f9;
}

.c13:active {
  background-color: #eaeaef;
}

.c13 .c8 {
  color: #32324d;
}

.c13 svg > g,
.c13 svg path {
  fill: #32324d;
}

.c15 {
  height: 2rem;
}

.c15[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c15[aria-disabled='true'] .c8 {
  color: #666687;
}

.c15[aria-disabled='true'] svg > g,.c15[aria-disabled='true'] svg path {
  fill: #666687;
}

.c15[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c15[aria-disabled='true']:active .c8 {
  color: #666687;
}

.c15[aria-disabled='true']:active svg > g,.c15[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c15:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c15:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c15 svg > g,
.c15 svg path {
  fill: #ffffff;
}

.c31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c33 {
  line-height: 0;
}

.c35 {
  border: none;
  border-radius: 4px;
  padding-bottom: 0.65625rem;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 0.65625rem;
  color: #32324d;
  font-weight: 400;
  font-size: 0.875rem;
  display: block;
  width: 100%;
  background: inherit;
}

.c35::-webkit-input-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c35::-moz-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c35:-ms-input-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c35::placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c35[aria-disabled='true'] {
  color: inherit;
}

.c35:focus {
  outline: none;
  box-shadow: none;
}

.c34 {
  border: 1px solid #dcdce4;
  border-radius: 4px;
  background: #ffffff;
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c34:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c42 {
  border: 1px solid #dcdce4;
  min-height: 2.5rem;
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c42[aria-disabled='true'] {
  color: #666687;
}

.c42:focus-visible {
  outline: none;
}

.c42:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c48 > svg {
  width: 0.375rem;
}

.c48 > svg > path {
  fill: #666687;
}

.c46 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c47 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c53[data-state='checked'] .c8 {
  font-weight: bold;
  color: #4945ff;
}

.c28 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  gap: 20px;
}

.c29 {
  grid-column: span 6;
  max-width: 100%;
}

.c0:focus-visible {
  outline: none;
}

.c37 {
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c37:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c39 {
  border: none;
  resize: none;
}

.c39::-webkit-input-placeholder {
  color: #8e8ea9;
  font-size: 0.875rem;
  color: #8e8ea9;
  opacity: 1;
}

.c39::-moz-placeholder {
  color: #8e8ea9;
  font-size: 0.875rem;
  color: #8e8ea9;
  opacity: 1;
}

.c39:-ms-input-placeholder {
  color: #8e8ea9;
  font-size: 0.875rem;
  color: #8e8ea9;
  opacity: 1;
}

.c39::placeholder {
  color: #8e8ea9;
  font-size: 0.875rem;
  color: #8e8ea9;
  opacity: 1;
}

.c39:focus-within {
  outline: none;
}

.c4 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #4945ff;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  gap: 8px;
  position: relative;
  outline: none;
}

.c3 svg {
  font-size: 0.625rem;
}

.c3 svg path {
  fill: #4945ff;
}

.c3:hover {
  color: #7b79ff;
}

.c3:active {
  color: #271fe0;
}

.c3:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c3:focus-visible {
  outline: none;
}

.c3:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c20 {
  margin-right: 24px;
}

.c20 svg {
  width: 2rem;
  height: 2rem;
}

.c23 {
  word-break: break-all;
}

@media (max-width:68.75rem) {
  .c29 {
    grid-column: span;
  }
}

@media (max-width:34.375rem) {
  .c29 {
    grid-column: span 12;
  }
}

<div>
  <main
    aria-labelledby="main-content-title"
    class="c0"
    id="main-content"
    tabindex="-1"
  >
    <form
      action="#"
      novalidate=""
    >
      <div
        style="height: 0px;"
      >
        <div
          class="c1"
          data-strapi-header="true"
        >
          <div
            class="c2"
          >
            <a
              aria-current="page"
              class="c3 active"
              href="/settings/transfer-tokens"
            >
              <svg
                fill="none"
                height="1rem"
                viewBox="0 0 24 24"
                width="1rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24 13.3a.2.2 0 0 1-.2.2H5.74l8.239 8.239a.2.2 0 0 1 0 .282L12.14 23.86a.2.2 0 0 1-.282 0L.14 12.14a.2.2 0 0 1 0-.282L11.86.14a.2.2 0 0 1 .282 0L13.98 1.98a.2.2 0 0 1 0 .282L5.74 10.5H23.8c.11 0 .2.09.2.2v2.6Z"
                  fill="#212134"
                />
              </svg>
              <span
                class="c4"
              >
                Back
              </span>
            </a>
          </div>
          <div
            class="c5"
          >
            <div
              class="c6 c7"
            >
              <h1
                class="c8 c9"
              >
                My super token
              </h1>
            </div>
            <div
              class="c10"
            >
              <button
                aria-disabled="false"
                class="c11 c10 c12 c13"
                name="regenerate"
                type="button"
              >
                <div
                  aria-hidden="true"
                  class=""
                >
                  <svg
                    fill="none"
                    height="1rem"
                    viewBox="0 0 24 24"
                    width="1rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M15.681 2.804A9.64 9.64 0 0 0 11.818 2C6.398 2 2 6.48 2 12c0 5.521 4.397 10 9.818 10 2.03 0 4.011-.641 5.67-1.835a9.987 9.987 0 0 0 3.589-4.831 1.117 1.117 0 0 0-.664-1.418 1.086 1.086 0 0 0-1.393.676 7.769 7.769 0 0 1-2.792 3.758 7.546 7.546 0 0 1-4.41 1.428V4.222h.002a7.492 7.492 0 0 1 3.003.625 7.61 7.61 0 0 1 2.5 1.762l.464.551-2.986 3.042a.186.186 0 0 0 .129.316H22V3.317a.188.188 0 0 0-.112-.172.179.179 0 0 0-.199.04l-2.355 2.4-.394-.468-.02-.02a9.791 9.791 0 0 0-3.239-2.293Zm-3.863 1.418V2v2.222Zm0 0v15.556c-4.216 0-7.636-3.484-7.636-7.778s3.42-7.777 7.636-7.778Z"
                      fill="#212134"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
                <span
                  class="c8 c14"
                >
                  Regenerate
                </span>
              </button>
              <button
                aria-disabled="false"
                class="c11 c10 c12 c15"
                type="submit"
              >
                <div
                  aria-hidden="true"
                  class=""
                >
                  <svg
                    fill="none"
                    height="1rem"
                    viewBox="0 0 24 24"
                    width="1rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M20.727 2.97a.2.2 0 0 1 .286 0l2.85 2.89a.2.2 0 0 1 0 .28L9.554 20.854a.2.2 0 0 1-.285 0l-9.13-9.243a.2.2 0 0 1 0-.281l2.85-2.892a.2.2 0 0 1 .284 0l6.14 6.209L20.726 2.97Z"
                      fill="#212134"
                    />
                  </svg>
                </div>
                <span
                  class="c8 c14"
                >
                  Save
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c16"
      >
        <div
          class="c17"
        >
          <div
            class="c18 c7"
          >
            <div
              class="c19 c7 c20"
            >
              <svg
                fill="none"
                height="1rem"
                viewBox="0 0 24 24"
                width="1rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 14h-4.34a6 6 0 1 1 0-4H23v4h-2v4h-4v-4ZM7 14a2 2 0 1 0 0-4 2 2 0 0 0 0 4Z"
                  fill="#8E8EA9"
                />
              </svg>
            </div>
            <div
              class="c21"
            >
              <div
                class="c7"
              >
                <span
                  class="c8 c22 c23"
                >
                  This token isn’t accessible anymore.
                </span>
              </div>
              <span
                class="c8 c24"
              >
                For security reasons, you can only see your token once.
              </span>
            </div>
          </div>
          <div
            class="c25"
          >
            <div
              class="c26"
            >
              <h2
                class="c8 c27"
              >
                Details
              </h2>
              <div
                class="c28"
              >
                <div
                  class="c29"
                >
                  <div>
                    <div
                      class=""
                    >
                      <div
                        class="c21"
                      >
                        <label
                          class="c8 c30 c31"
                          for=":r1:"
                        >
                          Name
                          <span
                            class="c8 c32 c33"
                          >
                            *
                          </span>
                        </label>
                        <div
                          class="c5 c34"
                        >
                          <input
                            aria-disabled="false"
                            aria-invalid="false"
                            aria-required="true"
                            class="c35"
                            id=":r1:"
                            name="name"
                            value="My super token"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="c29"
                >
                  <div
                    class=""
                  >
                    <div
                      class="c21"
                    >
                      <label
                        class="c8 c30 c31"
                        for="description"
                      >
                        Description
                      </label>
                      <div
                        class="c36 c37"
                      >
                        <textarea
                          aria-invalid="false"
                          aria-required="false"
                          class="c38 c39"
                          font-size="2"
                          id="description"
                        >
                          This describe my super token
                        </textarea>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="c29"
                >
                  <div
                    class=""
                  >
                    <div
                      class="c21"
                    >
                      <label
                        class="c8 c30 c31"
                        for=":r5:"
                      >
                        Token duration
                        <span
                          class="c8 c32 c33"
                        >
                          *
                        </span>
                      </label>
                      <div
                        aria-autocomplete="none"
                        aria-controls="radix-:r8:"
                        aria-describedby=":r5:-hint :r5:-error"
                        aria-disabled="true"
                        aria-expanded="false"
                        aria-label="Token duration"
                        aria-required="true"
                        class="c40 c41 c42"
                        data-disabled=""
                        data-state="closed"
                        dir="ltr"
                        id=":r5:"
                        overflow="hidden"
                        role="combobox"
                      >
                        <span
                          class="c43 c44"
                        >
                          <span
                            class="c8 c45 c46"
                          >
                            <span
                              class="c47"
                            >
                              <span>
                                Select
                              </span>
                            </span>
                          </span>
                        </span>
                        <span
                          class="c44"
                        >
                          <span
                            aria-hidden="true"
                            class="c48"
                          >
                            <svg
                              fill="none"
                              height="1rem"
                              viewBox="0 0 14 8"
                              width="1rem"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                clip-rule="evenodd"
                                d="M14 .889a.86.86 0 0 1-.26.625L7.615 7.736A.834.834 0 0 1 7 8a.834.834 0 0 1-.615-.264L.26 1.514A.861.861 0 0 1 0 .889c0-.24.087-.45.26-.625A.834.834 0 0 1 .875 0h12.25c.237 0 .442.088.615.264a.86.86 0 0 1 .26.625Z"
                                fill="#32324D"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                  <span
                    class="c8 c49"
                  >
                    Expiration date: Unlimited
                  </span>
                </div>
                <div
                  class="c29"
                >
                  <div
                    class=""
                  >
                    <div
                      class="c21"
                    >
                      <label
                        class="c8 c30 c31"
                        for=":r9:"
                      >
                        Token type
                        <span
                          class="c8 c32 c33"
                        >
                          *
                        </span>
                      </label>
                      <div
                        aria-autocomplete="none"
                        aria-controls="radix-:rc:"
                        aria-describedby=":r9:-hint :r9:-error"
                        aria-disabled="false"
                        aria-expanded="false"
                        aria-label="Token type"
                        aria-required="true"
                        class="c50 c41 c42"
                        data-state="closed"
                        dir="ltr"
                        id=":r9:"
                        overflow="hidden"
                        role="combobox"
                        tabindex="0"
                      >
                        <span
                          class="c43 c44"
                        >
                          <span
                            class="c8 c45 c46"
                          >
                            <span
                              class="c47"
                            >
                              <span>
                                Select
                              </span>
                            </span>
                          </span>
                        </span>
                        <span
                          class="c44"
                        >
                          <span
                            aria-hidden="true"
                            class="c48"
                          >
                            <svg
                              fill="none"
                              height="1rem"
                              viewBox="0 0 14 8"
                              width="1rem"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                clip-rule="evenodd"
                                d="M14 .889a.86.86 0 0 1-.26.625L7.615 7.736A.834.834 0 0 1 7 8a.834.834 0 0 1-.615-.264L.26 1.514A.861.861 0 0 1 0 .889c0-.24.087-.45.26-.625A.834.834 0 0 1 .875 0h12.25c.237 0 .442.088.615.264a.86.86 0 0 1 .26.625Z"
                                fill="#32324D"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </main>
  <div
    class="c51"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
